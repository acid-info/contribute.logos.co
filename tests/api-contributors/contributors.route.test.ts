import { describe, it, expect, vi } from 'vitest'
import mocks, { MOCK_ORGS } from './mocks'

function windowIso(days: number) {
  const until = new Date('2025-08-12T00:00:00Z')
  const since = new Date(until.getTime() - days * 24 * 60 * 60 * 1000)
  return { since: since.toISOString(), until: until.toISOString() }
}

vi.mock('@/lib/github', () => mocks)

describe('GET /api/contributors', () => {
  it('returns external contributors with latest activity (mocked)', async () => {
    const orgs = MOCK_ORGS.join(',')
    const { since, until } = windowIso(192)
    const url = `http://localhost/api/contributors?orgs=${encodeURIComponent(
      orgs
    )}&since=${encodeURIComponent(since)}&until=${encodeURIComponent(
      until
    )}&limitReposPerOrg=2&maxPrPages=2&maxReviewFetches=10&commitStrategy=search&debug=true`
    const req = new Request(url)
    const { GET: GETContributors } = await import('../../app/api/contributors/route')
    const res = await GETContributors(req)
    expect(res.status).toBe(200)
    const meta = res.headers.get('x-meta')
    expect(meta).toBeTruthy()
    const json = (await res.json()) as any[]
    expect(json.slice(0, 2)).toMatchSnapshot('contributors-list-sample')
    expect(JSON.parse(meta!)).toMatchSnapshot('contributors-list-meta')
  })
})
