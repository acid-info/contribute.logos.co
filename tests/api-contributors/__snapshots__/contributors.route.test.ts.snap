// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GET /api/contributors > returns external contributors with latest activity (mocked) > contributors-list-meta 1`] = `
{
  "commitCount": 2,
  "durationMs": 1,
  "errors": [],
  "orgs": [
    "org-mock-c",
    "org-mock-d",
  ],
  "prCount": 0,
  "repos": 0,
  "reviewCount": 0,
  "since": "2025-02-01T00:00:00.000Z",
  "until": "2025-08-12T00:00:00.000Z",
}
`;

exports[`GET /api/contributors > returns external contributors with latest activity (mocked) > contributors-list-sample 1`] = `
[
  {
    "contributionCount": 1,
    "internalFilter": "public-membership-only",
    "latest": {
      "date": "2025-07-02T00:00:00Z",
      "link": "https://github.com/contoso/server/commit/abc",
      "repo": "contoso/server",
      "type": "COMMIT",
    },
    "login": "author-mock",
    "profileUrl": "https://github.com/author-mock",
  },
]
`;
